src/transformers/trainer.py

```diff
@@ -1760,6 +1768,13 @@ def _inner_training_loop(
         if delay_optimizer_creation:
             self.create_optimizer_and_scheduler(num_training_steps=max_steps)
 
+        # prepare using `accelerator` prepare
+        model, self.optimizer, self.lr_scheduler = self.accelerator.prepare(
+            self.model, self.optimizer, self.lr_scheduler
+        )
+        self.accelerator.print(f"{model=}\n{self.optimizer=}\n{self.lr_scheduler=}")
+        self.accelerator.print(f"{model.forward=}")
```

Comment:
Are all these `print` debugging statements or meant to be logged? If they're meant to be logged, let's use the `logger` and put it under `debug`